src/components/PokemonCard/index.test.tsx

```diff
@@ -19,137 +14,51 @@ const pokemon = {
   weight: 100,
 };
 
-describe('Pokemon page', () => {
-  global.fetch = vi.fn(() =>
-    Promise.resolve({
-      json: () =>
-        Promise.resolve({
-          data: { pokemon },
-        }),
-    })
-  ) as unknown as typeof global.fetch;
-
-  const globalRoutes = [
-    {
-      path: '/',
-      element: (
-        <Context.Provider
-          value={{
-            pokemons: [{ name: 'Ivysaur', url: 'https://pokeapi.co/api/v2/pokemon/2/' }],
-            query: '',
-            setQuery: () => {},
-            setPokemons: () => {},
-          }}
-        >
-          <App />
-        </Context.Provider>
-      ),
-      errorElement: <ErrorPage />,
-      children: [
-        {
-          path: 'search/:pokemonId',
-          element: <PokemonPage />,
-        },
-      ],
-    },
-  ];
-
+describe('Pokemon card', () => {
   it('Check that a loading indicator is displayed while fetching data', async () => {
-    const routes = [
-      {
-        path: '/',
-        element: (
-          <Context.Provider
-            value={{
-              pokemons: [{ name: 'Ivysaur', url: 'https://pokeapi.co/api/v2/pokemon/2/' }],
-              query: '',
-              setQuery: () => {},
-              setPokemons: () => {},
-            }}
-          >
-            <PokemonCard pokemon={pokemon} loading />
-          </Context.Provider>
-        ),
-      },
-    ];
-
-    const router = createMemoryRouter(routes, {
-      initialEntries: ['/'],
-    });
-
-    const { container } = render(
-      <Theme>
-        <RouterProvider router={router} />
-      </Theme>
-    );
+    const { container } = renderWithProviders(<PokemonCard pokemon={pokemon} loading />);
 
     waitFor(() => {
-      const spinner = container.querySelector('.sc-bdfCDU.jUNipm'); // Classname for spinner
+      const spinner = container.querySelector('.kZUWme'); // Classname for spinner
```

Comment:
Sorry, but this is hardcode. You need find another way to get this spinner.

Reply:
I already fixed that up, didn't push that